src/intersection/first-solution/intersection.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Creates an array of values that are in both the first and the second arrays.
+ *
+ * @param {Array} [firstArr=[]] - First array to be compared for repeated values.
+ * @param {Array} [secondArr=[]] - Second array to be compared for repeated values.
+ * @returns {Array} Returns the new array with only intersected values.
+ */
+
+export const arrayIntersection = (firstArr = [], secondArr = []) => {
+  let newArrayIntersected = [];
+  // Test if the arguments are arrays and make sure that arguments are passed
+  if (!Array.isArray(firstArr) || !Array.isArray(secondArr)) {
+    throw new TypeError('The arguments must be arrays');
+  } else if (!firstArr.length || !secondArr.length) {
+    throw new Error('Must pas two arrays to intercept the values');
```

Comment:
the purpose of the function is to intersect values and in order to do that, I  have create the code to work only for intersection, if one argument is missing it will not work, of course I could make it more flexible but, that is according of the behavior that you want it to have